src/containers/signup/Signup.tsx

```diff
@@ -1,20 +1,22 @@
 import React from 'react';
 import { Helmet } from 'react-helmet';
 import { Button, Form, Input, Typography } from 'antd';
-import { signup } from '../../auth/authAPI';
 import { Link } from 'react-router-dom';
+import { signup } from '../../auth/ducks/thunks';
+import { useDispatch } from 'react-redux';
 const { Title, Paragraph } = Typography;
 
 const Signup: React.FC = () => {
+  const dispatch = useDispatch();
   const onFinish = (values: any) => {
-    // TODO: what if backend says the values are invalid? need to handle this
```

Comment:
What happens in this case now?

Reply:
The AsyncRequest utility file will dictate the state of any action defined as an AsyncRequest. It defines `notStarted`, `loading`, `loaded`, and `failed`. So depending on the state of the user authentication async request, we'll conditionally render data. The backend-scaffold also currently only says "Malformed JSON request" so I can't yet implement an Error type on the AsyncRequest. The util optionally allows an Error data payload to be stored, so we should be able to strongly type the backend responses to populate errors on form inputs.